linkerd/tls/src/server/mod.rs

```diff
@@ -181,22 +203,20 @@ where
             None => {
                 let peer = Conditional::None(NoServerTls::Disabled);
                 let svc = new_accept.new_service((peer, target));
-                Box::pin(svc.oneshot(EitherIo::Left(io.into())).err_into::<Error>())
+                Box::pin(
+                    svc.oneshot(EitherIo::Right(EitherIo::Left(io)))
+                        .err_into::<Error>(),
+                )
             }
         }
     }
 }
 
-async fn detect<I>(
-    mut io: I,
-    tls_config: Config,
-    LocalId(local_id): LocalId,
-) -> io::Result<(ConditionalServerTls, Io<I>)>
+// Peek or buffer the provided stream to determine an SNI value.
```

Comment:
*very* nitpicky (sorry, this is dumb): 
```suggestion
/// Peek or buffer the provided stream to determine an SNI value.
```